sdc/rabbit/async_consumer.py

```diff
@@ -0,0 +1,388 @@
+import logging
+import time
+
+import pika
+from structlog import wrap_logger
+
+logger = wrap_logger(logging.getLogger('__name__'))
+
+
+class AsyncConsumer():
+    """This is an example consumer that will handle unexpected interactions
+    with RabbitMQ such as channel and connection closures.
+
+    If RabbitMQ closes the connection, it will reopen it. You should
+    look at the output, as there are limited reasons why the connection may
+    be closed, which usually are tied to permission related issues or
+    socket timeouts.
+
+    If the channel is closed, it will indicate a problem with one of the
+    commands that were issued and that should surface in the output as well.
+
+    """
+
+    def __init__(self,
+                 durable_queue,
+                 exchange,
+                 exchange_type,
+                 rabbit_queue,
+                 rabbit_urls):
+        """Create a new instance of the AsyncConsumer class.
+
+        :param durable_queue: Boolean specifying whether queue is durable
+        :param exchange: RabbitMQ exchange name
+        :param exchange_type: RabbitMQ exchange type
+        :param rabbit_queue: RabbitMQ queue name
+        :param rabbit_urls: List of rabbit urls
+
+        :returns Object of type AsyncConsumer
+        :rtype AsyncConsumer
+
+        """
+        self._exchange = exchange
+        self._exchange_type = exchange_type
+        self._durable_queue = durable_queue
+        self._queue = rabbit_queue
+        self._rabbit_urls = rabbit_urls
+        self._rabbit_quarantine_queue = 'async_consumer_quarantine'
```

Comment:
It's not. I've taken this out and put a required parameter instead. This can be passed from the calling code so is configurable.